src/Bridge/Symfony/ContainerServiceProvider.php

```diff
@@ -6,26 +6,39 @@
 
 use Rector\Core\Configuration\RectorConfigProvider;
 use Rector\Core\Exception\ShouldNotHappenException;
+use Symfony\Component\DependencyInjection\Container;
 use Webmozart\Assert\Assert;
 
 final class ContainerServiceProvider
 {
+    private ?object $container = null;
+
     public function __construct(
         private readonly RectorConfigProvider $rectorConfigProvider
     ) {
     }
 
-    public function provideByName(string $serviceName): object
+    private function getSymfonyContainer(): object
     {
-        $symfonyContainerPhp = $this->rectorConfigProvider->getSymfonyContainerPhp();
-        Assert::fileExists($symfonyContainerPhp);
+        if ($this->container === null) {
+            $symfonyContainerPhp = $this->rectorConfigProvider->getSymfonyContainerPhp();
+            Assert::fileExists($symfonyContainerPhp);
 
-        $container = require_once $symfonyContainerPhp;
+            $container = require $symfonyContainerPhp;
 
-        // this allows older Symfony versions, e.g. 2.8 did not have the PSR yet
-        Assert::isInstanceOf($container, 'Symfony\Component\DependencyInjection\Container');
+            // this allows older Symfony versions, e.g. 2.8 did not have the PSR yet
+            Assert::isInstanceOf($container, 'Symfony\Component\DependencyInjection\Container');
+            $this->container = $container;
+        }
 
-        if (! $container->has($serviceName)) {
+        return $this->container;
+    }
+
+    public function provideByName(string $serviceName): object
+    {
+        /** @var Container $container */
+        $container = $this->getSymfonyContainer();
+        if (!$container->has($serviceName)) {
```

Comment:
```suggestion
        if (! $container->has($serviceName)) {
```
I think it should be like this but the pipeline will also mention this